src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2417,6 +2417,16 @@ private function registerNotifierConfiguration(array $config, ContainerBuilder $
             $container->removeDefinition($classToServices[MercureTransportFactory::class]);
         }
 
+        if (ContainerBuilder::willBeAvailable('symfony/fake-chat-notifier', FakeChatTransportFactory::class, array_merge($parentPackages, ['symfony/mailer']))) {
```

Comment:
Not sure this line is correct 🤔 

I only want to run this code if `fake-chat-notifier` with `FakeChatTransportFactory::class` and `symfony/mailer` is installed. Am I using `willBeAvailable` method correct?

Friendly ping @stof

Reply:
The part I'm not sure is the `array_merge`. I don't remember whether passing multiple parent packages requires all of them or at least one of them. /cc @nicolas-grekas

Reply:
Regarding this code:
https://github.com/symfony/symfony/blob/ddb48bb39123b18b0f73f1e19fb0747aed8590b0/src/Symfony/Component/DependencyInjection/ContainerBuilder.php#L1469-L1495

I would say it just checks if at least one of them is available 🤔

Reply:
meaning that if you want to ensure that several parent packages are all installed, you should do that as `ContainerBuilder::willBeAvailable(..., $parent1) && ContainerBuilder::willBeAvailable(..., $parent2)` :smile:

Reply:
Yes, but in this case I am wondering if the previous used one by mercure is working as expected and needs to do the call twice too 🤔 

cc @mtarld

Reply:
Yes indeed, the call is done twice for mercure

Reply:
Yes but you are using `$parentPackages`, but I think you need to achieve, that both, notifier and framework-bundle need to be available, right? But this would not work.

To be more clear, I think we need to change your code to:
```diff
diff --git a/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php b/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php
index 4d4749a2f1..f36702d254 100644
--- a/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php
+++ b/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php
@@ -2410,10 +2410,16 @@ class FrameworkExtension extends Extension
             }
         }

-        if (ContainerBuilder::willBeAvailable('symfony/mercure-notifier', MercureTransportFactory::class, $parentPackages) && ContainerBuilder::willBeAvailable('symfony/mercure-bundle', MercureBundle::class, $parentPackages)) {
+        if (ContainerBuilder::willBeAvailable('symfony/mercure-notifier', MercureTransportFactory::class, ['symfony/framework-bundle'])
+            && ContainerBuilder::willBeAvailable('symfony/mercure-notifier', MercureTransportFactory::class, ['symfony/notifier'])
+            && ContainerBuilder::willBeAvailable('symfony/mercure-bundle', MercureBundle::class, ['symfony/framework-bundle'])
+            && ContainerBuilder::willBeAvailable('symfony/mercure-bundle', MercureBundle::class, ['symfony/notifier'])
+        ) {
             $container->getDefinition($classToServices[MercureTransportFactory::class])
                 ->replaceArgument('$registry', new Reference(HubRegistry::class));
-        } elseif (ContainerBuilder::willBeAvailable('symfony/mercure-notifier', MercureTransportFactory::class, $parentPackages)) {
+        } elseif (ContainerBuilder::willBeAvailable('symfony/mercure-notifier', MercureTransportFactory::class, ['symfony/framework-bundle'])
+            && ContainerBuilder::willBeAvailable('symfony/mercure-notifier', MercureTransportFactory::class, ['symfony/notifier'])
+        ) {
             $container->removeDefinition($classToServices[MercureTransportFactory::class]);
:
```

if **yes** maybe it makes sense to either extend the `willBeAvailable` method to a 4th parameter `bool $all = false` or create a `allWillBeAvailable` and deprecated `$parentPackages` to be an array in `willBeAvailable` method.

cc @nicolas-grekas 

PS: I am unsure because I use this method the first time 😸